_posts/2017-04-25-the-everchanging-software-process.md

```diff
@@ -0,0 +1,37 @@
+ 
+I started to work on the Harja project almost 3 years ago. The project itself is a large web based open-source service ([Harja@GitHub](https://github.com/finnishtransportagency/harja/)) ordered by the [Finnish Transport Agency](http://www.liikennevirasto.fi/web/en) meant for planning and monitoring road, waterway and railway maintenance and care. At first we assumed a pretty much "classic" scrum model as our very basic outline. This was pretty much because most of us we're quire familiar with it and had positive experience in previous projects. Pretty soon we started to notice things here and there that didn't make sense and were starting to feel as a pain, such as over booked sprints, badly defined tasks, review sessions that didn't really give much value.
+ 
+From the very beginning we've thought our processes and ways of working as "snapshot in time", which serves the current needs of the team, project, service and customer. Obviously all of these are volatile by nature and therefore we found that we needed to have a mechanism to meet these changes time and a time again. 
+ 
+For us, one of the key solutions came from regularly held agile retrospectives. Not that this would be any kind of a revolutionary method, but the idea behind it gave us a chance to iteratively monitor and change our way of working. From very early days of our project, we've followed the following mantra:
+1. Reflect on the past period of time: Observer what doesn't work and needs to be changed and what does and has to be protected and nurtured.
+2. Design the changes that you as a team agree that would be beneficial. This phase requires extra care, so that the actions are very thoroughly defined, they have set responsible people and schedule, conditions which have to be met and finally a defined reason why this action will be taken. (This phase is heavily affected by [S.M.A.R.T. goals](https://www.mindtools.com/pages/article/smart-goals.htm))
+3. Take the actions and bring implement them. At this phase the key is to incorporate the changes just you would for example any other development issue. For us this meant that the development actions simply written as a JIRA ticket and placed on our board.
+4. In the next retrospective, go through all of the actions and see:
+- if they've had the wanted effect
+- if they require more work
+- if they we're a bad idea just scrap them, for there can be no love for ineffective actions
+ 
+As mentioned before, this is not by far close to any kind of rocket science. Though this kind of way thinking and working has affected a whole range of aspects in our project work, from building and greasing parempi? our CI-pipeline, improving our quality assurance to finding better ways to communicate with our client. For more concrete examples, that have gone through this very process you find from from the following blog posts:
```

Comment:
"kind of way thinking" ylimääräinen "way"

"greasing parempi?" => ehkä "...from building our CI-pipeline and making it more robust to improving our quality assurance and finding..."

"you find" => "you can find"